### ISSUE TITLE
Dialog sizing issue on some Linux desktop environments

### ISSUE THREAD

**Issue by aglkm** [sparrowwallet/sparrow#1157]
I've upgraded from 1.7.1 to 1.8.0 and see the following when trying to create new wallet:

![Screenshot from 2023-11-10 17-27-59](https://github.com/sparrowwallet/sparrow/assets/39521015/be2a8a35-1ff1-4235-9e74-f877104a59c4)

The only message I see in the console is the following:
(Sparrow:6487): Gdk-WARNING **: 17:27:07.369: XSetErrorHandler() called with a GDK error trap pushed. Don't do that.

When reverting back to 1.7.1 everything works well.

OS: Debian GNU/Linux trixie/sid x86_64
Kernel: 6.5.0-3-amd64
DE: GNOME 44.5

**Comment by craigraw** [comment]
Can you try v1.7.9?

**Comment by aglkm** [comment]
Just tried 1.7.9. There's no such issue there.

**Comment by craigraw** [comment]
Ok - the only change I can think of that would affect this is setting the ownership of the dialogs you are using (https://github.com/sparrowwallet/sparrow/commit/31539a27ac7bf3548d529bf6b9ff61ef5503c210). Looks like a GNOME bug tbh, since there should be no problem with doing this. I have tested it without issues on Ubuntu and Debian 11.3.

**Comment by onlineapps-cloud** [comment]
hi, same error after upgrade sparrow wallet
![image](https://github.com/sparrowwallet/sparrow/assets/69077240/e5c480b7-c426-425b-99fd-19851c1f392f)

**Comment by onlineapps-cloud** [comment]
> Just tried 1.7.9. There's no such issue there.

i can confirm, on this version, all work as expected.

**Comment by craigraw** [comment]
Any errors in sparrow.log? See Help menu

**Comment by onlineapps-cloud** [comment]
```
2023-11-13 14:06:58,340 WARN [Thread-15] c.s.s.n.ExchangeSource [null:-1] Error retrieving currency rates (Server returned HTTP response code: 403 for URL: https://api.coinbase.com/v2/exchange-rates?currency=BTC)
2023-11-13 14:07:15,342 WARN [Thread-18] c.s.s.n.ExchangeSource [null:-1] Error retrieving currency rates (Server returned HTTP response code: 403 for URL: https://api.coinbase.com/v2/exchange-rates?currency=BTC)
2023-11-13 14:07:15,376 ERROR [JavaFX Application Thread] c.s.s.p.GeneralPreferencesController [null:-1] Error retrieving currencies
java.lang.NullPointerException: Cannot read field "rates" because "com.sparrowwallet.sparrow.net.ExchangeSource$2.getRates().data" is null
	at com.sparrowwallet.sparrow@1.8.0/com.sparrowwallet.sparrow.net.ExchangeSource$2.getSupportedCurrencies(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.0/com.sparrowwallet.sparrow.net.ExchangeSource$CurrenciesService$1.call(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.0/com.sparrowwallet.sparrow.net.ExchangeSource$CurrenciesService$1.call(Unknown Source)
	at javafx.graphics@18/javafx.concurrent.Task$TaskCallable.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at javafx.graphics@18/javafx.concurrent.Service.lambda$executeTask$6(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at javafx.graphics@18/javafx.concurrent.Service.lambda$executeTask$7(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
2023-11-13 14:07:19,102 WARN [Thread-20] c.s.s.n.ExchangeSource [null:-1] Error retrieving currency rates (Server returned HTTP response code: 403 for URL: https://api.coinbase.com/v2/exchange-rates?currency=BTC)

```

**Comment by aglkm** [comment]
I switched my DE to try the issue on Plasma 5.27.9 on my system and re-created there as well. The graphic glitches a bit different there, but I can always re-create by the following steps:
1. File - New Wallet - Create Wallet
2. Click on any type of wallet to create
3. Close the dialogue if not re-created and try step 2 again several times

![sp_issue](https://github.com/sparrowwallet/sparrow/assets/39521015/a34ccbf0-1486-4b05-b7be-f5073a1b6b9c)

**Comment by craigraw** [comment]
Ok, I've managed to reproduce it. Looking into this further...

**Comment by craigraw** [comment]
It appears to be some kind of a thread race issue on these Linux DEs causing some dialogs to resize to their minimum possible size. I have fixed these occurrences in 9391a397, either by setting a minimum size explicitly or making the dialog resizable, which seems to work as well.

Please can you test the beta build at https://github.com/craigraw/beta/releases/tag/1.8.1-plasmafix. 

@aglkm I would particularly appreciate testing on GNOME 44.5, which I don't currently have.

**Comment by onlineapps-cloud** [comment]
good news, on my sistem work as expected
![image](https://github.com/sparrowwallet/sparrow/assets/69077240/4aa9b6ae-bde0-422a-a8e2-3d3c40499d14)
thanks for quick fixing.

**Comment by aglkm** [comment]
Fixed in GNOME 44.5. Thank you.

**Comment by craigraw** [comment]
Great, thank you for the quick testing. Closing this off, but if you see any further issues, please comment here.
